### PR TITLE
move RTPS to dedicated px4fmu-v{3,4,4pro,5}, posix, sdflight builds

### PR DESCRIPTION
**Pull request by dagar**
RTPS and micro-cdr now only included and required in the following builds

    - nuttx_px4fmu-v3_rtps
    - nuttx_px4fmu-v4_rtps
    - nuttx_px4fmu-v4pro_rtps
    - nuttx_px4fmu-v5_rtps

    - posix_sitl_rtps

    - posix_eagle_rtps
    - posix_excelsior_rtps

### CODE REVIEW

**Comment by dagar** [comment]
No, that's been failing intermittently.

File: cmake/configs/nuttx_px4fmu-v3_rtps.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+include(configs/nuttx_px4fmu-v3_default)
+include(common/rtps)
+
+list(APPEND config_module_list
+	# micro RTPS
+	modules/micrortps_bridge/micrortps_client
+	lib/micro-CDR
+)
+
+set(config_rtps_send_topics
```
Comment (bkueng) [inline cmake/configs/nuttx_px4fmu-v3_rtps.cmake]: Can we move the set of topics to `common/rtps`? Will be easier to manage.

File: Makefile [hunk 2]
```diff
@@ -209,6 +213,14 @@ alt_firmware: \
 	check_s2740vc-v1_default \
 	sizes
 
+# builds with RTPS
+check_rtps: \
```
Comment (bkueng) [inline Makefile]: Is the plan to add this to CI? If so, we'll need to check the overhead of building all targets.

File: cmake/configs/nuttx_px4fmu-v3_rtps.cmake (hunk 1 above)
Comment (dagar) [reply cmake/configs/nuttx_px4fmu-v3_rtps.cmake]: That sounds fine to me, but I don't think we know how this is actually intended to be used.

File: Makefile (hunk 2 above)
Comment (dagar) [reply Makefile]: Yes, this is for CI.

**Comment by dagar** [comment]
Apologies for continuing to change this, but what do you think about everything living within modules/micrortps_bridge? micro-CDR, generation, topic lists, etc

Little things like this will help as we work towards large restructuring for modularity (https://github.com/PX4/Firmware/issues/8127, https://github.com/PX4/Firmware/pull/8139).

See updated push.

**Comment by mcharleb** [comment]
Looks good. I'd still like to be able to use the generated agent code without having to sync and build PX4 to get it.

**Comment by dagar** [comment]
We could have Firmware output a library (with a cmake build system) that's automatically pushed to another github repo. That generated library could be versioned to match Firmware.

**Comment by dagar** [comment]
Good to merge?

**Comment by dagar** [comment]
Merging now to get it added to semaphoreci. I suspect this will go through a few more iterations once we figure out what to do with the generated agent code and the directory structure refactor.

**Pull request merged**